server/src/main/java/com/google/finapp/SpannerDaoImpl.java

```diff
@@ -155,22 +148,19 @@ public void createCustomerRole(
               });
       return accountBalancesBuilder.build();
     } catch (SpannerException e) {
-      // filter for IllegalArgumentExceptions thrown in lambda function above
+      // filter for StatusException thrown in lambda function above
       Throwable cause = e.getCause();
-      if (cause instanceof IllegalArgumentException) {
-        throw new IllegalArgumentException(cause.getMessage());
+      if (cause instanceof StatusException) {
+        throw (StatusException) cause;
       }
       throw new SpannerDaoException(e);
     }
   }
 
   @Override
   public BigDecimal createTransactionForAccount(
-      ByteArray accountId, BigDecimal amount, boolean isCredit) throws SpannerDaoException {
-    if (amount.signum() == -1) {
-      throw new IllegalArgumentException(
-          String.format("Amount transferred cannot be negative. amount: %s", amount.toString()));
-    }
+      ByteArray accountId, BigDecimal amount, boolean isCredit)
```

Comment:
Same question about `amount` as above.

Reply:
Resolving this and we can continue the conversation above.